### PR TITLE
Added async collection, wrapped up async transaction

### PR DESCRIPTION
**Pull request by JigarJoshi**
- Implemented async collection.
- Implemented transaction methods in async database.
- Modified tests.
#27

### CODE REVIEW

File: client/src/main/java/com/tigrisdata/db/client/service/StandardTigrisCollection.java [hunk 1]
```diff
@@ -50,44 +52,28 @@
...
-      Iterator<Api.ReadResponse> readResponseIterator = stub.read(readRequestBuilder.build());
+      Api.ReadRequest readRequest =
+          toReadRequest(databaseName, collectionName, filter, fields, readRequestOptions);
+      Iterator<Api.ReadResponse> readResponseIterator = stub.read(readRequest);
 
       Function<Api.ReadResponse, T> converter =
           readResponse -> {
             try {
-              return objectMapper.readValue(
+              return Utilities.OBJECT_MAPPER.readValue(
```
Comment (mattayres) [inline client/src/main/java/com/tigrisdata/db/client/service/StandardTigrisCollection.java:76]: I'd avoid referencing a static instance and prefer passing in `objectMapper` to the constructor for future flexibility. By default they can all be passed the same instance for the same effect.

File: client/src/main/java/com/tigrisdata/db/client/service/StandardTigrisAsyncCollection.java [hunk 2]
```diff
@@ -45,72 +74,199 @@ public void read(
...
       ReadRequestOptions readRequestOptions,
       TigrisDBAsyncReader<T> reader)
-      throws TigrisDBException {}
+      throws TigrisDBException {
+    try {
+      Api.ReadRequest readRequest =
+          toReadRequest(databaseName, collectionName, filter, fields, readRequestOptions);
+      stub.read(readRequest, new ReadManyResponseObserverAdapter<>(reader, collectionTypeClass));
+    } catch (JsonProcessingException jsonProcessingException) {
+      throw new TigrisDBException("Failed to read", jsonProcessingException);
```
[the remark this answers is not in the document]
Comment (JigarJoshi) [reply client/src/main/java/com/tigrisdata/db/client/service/StandardTigrisAsyncCollection.java]: updated the user facing exposure of this in all places.

File: client/src/main/java/com/tigrisdata/db/client/service/StandardTigrisAsyncCollection.java [hunk 3]
```diff
@@ -45,72 +74,199 @@ public void read(
...
       List<T> documents, InsertRequestOptions insertRequestOptions) throws TigrisDBException {
-    return null;
+    try {
+      Api.InsertRequest insertRequest =
+          toInsertRequest(databaseName, collectionName, documents, insertRequestOptions);
+      ListenableFuture<Api.InsertResponse> insertResponseListenableFuture =
+          futureStub.insert(insertRequest);
+      return Utilities.transformFuture(
+          insertResponseListenableFuture,
+          input -> new InsertResponse(new TigrisDBResponse("inserted")),
```
[the remark this answers is not in the document]
Comment (JigarJoshi) [reply client/src/main/java/com/tigrisdata/db/client/service/StandardTigrisAsyncCollection.java]: updated thank you.

File: client/src/main/java/com/tigrisdata/db/client/service/StandardTigrisCollection.java (hunk 1 above)
Comment (JigarJoshi) [reply client/src/main/java/com/tigrisdata/db/client/service/StandardTigrisCollection.java]: removed the static reference and added ability to pass in. I will revisit the visibility of some of these methods (For example toJSON() on filter shouldn't be public facing.

Comment (JigarJoshi) [reply client/src/main/java/com/tigrisdata/db/client/service/StandardTigrisCollection.java]: Thanks @mattayres

**Comment by JigarJoshi** [comment]
updated client with server API changes. (listCollections and listDatabase)

**Review by himank — APPROVED** [review]

**Review by mattayres — APPROVED** [review]

**Pull request merged**